staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication.go

```diff
@@ -57,6 +57,11 @@ func (a *AuthenticationInfoResolverDelegator) ClientConfigForService(serviceName
 
 type defaultAuthenticationInfoResolver struct {
 	kubeconfig clientcmdapi.Config
+
+	// skipInCluster causes the defaultAuthenticationInfoResolver skip the in-cluster config check.
+	// This is useful only for unit testing when it is done inside the actual cluster where the existing in-cluster config will cause
+	// the unit tests to give incorrect results.
+	skipInCluster bool
```

Comment:
Given the update below, is this still needed?

Reply:
@deads2k I think so as this calls `rest.InClusterConfig()` directly.